hw/drivers/chg_ctrl/da1469x_charger/src/da1469x_charger_shell.c

```diff
@@ -0,0 +1,970 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <os/mynewt.h>
+
+#if MYNEWT_VAL(DA1469X_CHARGER_CLI)
+
+#include <console/console.h>
+#include <shell/shell.h>
+#include <da1469x_charger/da1469x_charger.h>
+#include <parse/parse.h>
+#include <stdio.h>
+#include <DA1469xAB.h>
+#if MYNEWT_VAL(SDADC_BATTERY)
+#include <sdadc_da1469x/sdadc_da1469x.h>
+#endif
+#if MYNEWT_VAL(GPADC_BATTERY)
+#include <gpadc_da1469x/gpadc_da1469x.h>
+#endif
+
+#if MYNEWT_VAL(SHELL_CMD_HELP)
+//#define HELP(a) &(a)
+//#else
+#define HELP(a) NULL
+#endif
+
+/*
+ * Struct for field name description.
+ * Used for argument validation and field decoding
+ */
+struct reg_field
+{
+    /* Name of field */
+    const char *fld_name;
+    /* Position of field in register (LSB position) */
+    uint8_t fld_pos;
+    bool fld_show_bits;
+    /* Mask of bit field to set or extract */
+    uint32_t fld_mask;
+    /* Function that will convert register value to descriptive string */
+    const char *(*fld_decode_value)(const struct reg_field *field,
+            uint32_t reg_val, char *buf);
+    /* Argument that could be used by above function */
+    const void *fld_arg;
+};
+
+/**
+* Register structure
+*/
+struct reg {
+    /* Address of the register */
+    volatile uint32_t *addr;
+    /* Name of the register */
+    const char *name;
+    /* Array of bit fields */
+    const struct reg_field *fields;
+};
+
+static const char *charger_state[] = {
+    "POWER_UP",
+    "INIT",
+    "DISABLED",
+    "PRE_CHARGE",
+    "CC_CHARGE",
+    "CV_CHARGE",
+    "END_OF_CHARGE",
+    "TDIE_PROT",
+    "TBAT_PROT",
+    "BYPASSED",
+    "ERROR",
+    NULL,
+};
+
+#if MYNEWT_VAL(DA1469X_CHARGER_CLI_DECODE)
+
+/* Change ix value to string from table.
+ * Table is NULL terminated. Access outside of range will result in
+ * returning ??? string
+ */
+static const char *val_decode_from_table(const char *const tab[], int ix)
```

Comment:
Coding style - the function type must be on a line by itself preceding the function.